packages/core/src/test/codewhisperer/commands/startSecurityScanOld.test.ts

```diff
@@ -0,0 +1,271 @@
+        const testWindow = getTestWindow()
+        testWindow.onDidShowMessage(message => {
+            if (message.message === stopScanMessage) {
+                message.selectItem(startSecurityScan.stopScanButton)
+            }
+        })
+        model.codeScanState.setToRunning()
+        const scanPromise = startSecurityScan.startSecurityScanOld(
+            mockSecurityPanelViewProvider,
+            editor,
+            createClient(),
+            extensionContext
+        )
+        await startSecurityScan.confirmStopSecurityScan()
+        await scanPromise
+        assert.ok(securityScanRenderSpy.notCalled)
+        assert.ok(securityScanStoppedErrorSpy.calledOnce)
+        const warnings = testWindow.shownMessages.filter(m => m.severity === SeverityLevel.Warning)
+        assert.ok(warnings.map(m => m.message).includes(stopScanMessage))
+    })
+
+    it('Should not stop security scan when not confirmed', async function () {
+        getFetchStubWithResponse({ status: 200, statusText: 'testing stub' })
+        const securityScanRenderSpy = sinon.spy(diagnosticsProvider, 'initSecurityScanRenderOld')
+        const securityScanStoppedErrorSpy = sinon.spy(model, 'CodeScanStoppedError')
+        const testWindow = getTestWindow()
+        testWindow.onDidShowMessage(message => {
+            if (message.message === stopScanMessage) {
+                message.selectItem(cancel)
+            }
+        })
+        model.codeScanState.setToRunning()
+        const scanPromise = startSecurityScan.startSecurityScanOld(
+            mockSecurityPanelViewProvider,
+            editor,
+            createClient(),
+            extensionContext
+        )
+        await startSecurityScan.confirmStopSecurityScan()
+        await scanPromise
+        assert.ok(securityScanRenderSpy.calledOnce)
+        assert.ok(securityScanStoppedErrorSpy.notCalled)
+        const warnings = testWindow.shownMessages.filter(m => m.severity === SeverityLevel.Warning)
+        assert.ok(warnings.map(m => m.message).includes(stopScanMessage))
+    })
+
+    it('Should highlight files after scan is completed', async function () {
+        if (semver.lt(vscode.version, '1.78.0')) {
+            this.skip()
+        }
+        const commandSpy = sinon.spy(vscode.commands, 'executeCommand')
+        getFetchStubWithResponse({ status: 200, statusText: 'testing stub' })
+        const testWindow = getTestWindow()
+        testWindow.onDidShowMessage(message => {
+            if (message.message.includes('Security scan completed')) {
+                message.selectItem(showScannedFilesMessage)
+            }
+        })
+        await startSecurityScan.startSecurityScanOld(
+            mockSecurityPanelViewProvider,
+            editor,
+            createClient(),
+            extensionContext
+        )
+        assert.ok(commandSpy.calledWith(codeScanLogsOutputChannelId))
+        assertTelemetry('codewhisperer_securityScan', {
```

Comment:
you could try wrapping this in a `waitUntil` so that it retries a few times, if there is some async work happening above